code/game/objects/items/weaponry.dm

```diff
@@ -603,20 +603,20 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
 	. += span_notice("You can activate the bust in-hand to swear or forswear a Hippocratic Oath! This has no effects except pacifism or bragging rights. Does not remove other sources of pacifism. Do not eat.")
 
 /obj/item/statuebust/hippocratic/equipped(mob/living/carbon/human/user, slot)
-	..()
+	. = ..()
 	if(!(slot & ITEM_SLOT_HANDS))
 		return
-	var/datum/atom_hud/our_hud = GLOB.huds[DATA_HUD_MEDICAL_ADVANCED]
-	our_hud.show_to(user)
+	if(!HAS_TRAIT(user, TRAIT_MEDICAL_HUD))
+		var/datum/atom_hud/our_hud = GLOB.huds[DATA_HUD_MEDICAL_ADVANCED]
+		our_hud.show_to(user)
 	ADD_TRAIT(user, TRAIT_MEDICAL_HUD, type)
 
 /obj/item/statuebust/hippocratic/dropped(mob/living/carbon/human/user)
-	..()
-	if(HAS_TRAIT_NOT_FROM(user, TRAIT_MEDICAL_HUD, type))
-		return
-	var/datum/atom_hud/our_hud = GLOB.huds[DATA_HUD_MEDICAL_ADVANCED]
-	our_hud.hide_from(user)
+	. = ..()
 	REMOVE_TRAIT(user, TRAIT_MEDICAL_HUD, type)
+	if(!HAS_TRAIT(user, TRAIT_MEDICAL_HUD))
+		var/datum/atom_hud/our_hud = GLOB.huds[DATA_HUD_MEDICAL_ADVANCED]
+		our_hud.hide_from(user)
```

Comment:
This is not necessary anyways, huds soft-track sources that they're being added and removed from, all you should need to do is show it and hide without checking `HAS_TRAIT`

Reply:
It doesn't appear to be checked at all, fiddling with it, if you put medhuds on, take the bust in your hand, then take your huds off, you lose your medhud despite the bust giving you the trait, due to 
https://github.com/tgstation/tgstation/blob/6e9584ee271f10fac09cdc3a11b9adb1b327e714/code/modules/clothing/glasses/hud.dm#L17-L23

We have to change how hud glasses work to basically work the same as what I'm doing here in order for it to work.
https://github.com/tgstation/tgstation/assets/53777086/141e61a1-a510-433c-89ea-a094a29fc60d

Reply:
If it's not working then that's a bug with atom huds in general because that's how it's supposed to work, we shouldn't have to change hudglasses at all

Reply:
The trait is actually worthless btw, it's not at all for checking the actual atom hud, it's for checking if you can use the shift-click examine stuff

However maybe we should use `signal_addtrait` and `signal_removetrait` to handle huds instead of what it does now